symbol.go

```diff
@@ -0,0 +1,159 @@
+package protocol
+
+type SymbolData struct {
+	Text   string      `json:"text"`
+	Detail string      `json:"detail,omitempty"`
+	Kind   SymbolKind  `json:"kind"`
```

Comment:
Would it make sense to add a "language" field to specify what programming language the symbol is defined in? I have wanted to add this field to `SymbolInformation` in LSP. IntelliJ uses different icons for Scala and Java symbols, which I find helpful when searching for names of classes that exist in both languages

![Screenshot 2021-01-20 at 13 42 29](https://user-images.githubusercontent.com/1408093/105176347-703e9800-5b25-11eb-9fbd-91faf5cc91f9.png)

I think it would be nice to change symbol icons in the Sourcegraph client to be more similar to how IntelliJ displays them. I can never remember what the VS Code symbols mean in Scala codebases

![Screenshot 2021-01-20 at 13 44 43](https://user-images.githubusercontent.com/1408093/105176601-bd226e80-5b25-11eb-85d3-65c1256dde05.png)

Reply:
I think it would make sense. Probably out of the scope of this PR, though.

Reply:
This struct is intended to mirror the LSIF spec, right?
The language in LSIF is stored on the document vertex, so it could be inferred from there, but it's not part of the symbol struct.

Reply:
I think there's opportunity to extend LSIF to include information that is missing from LSP. The workspace/symbol request in LSP has several limitations that make it impossible to provide a rich search experience compared to IntelliJ. For example, symbol results are unnecessarily expensive to compute because you need to include the exact file location of the symbol definition, instead of being lazily loadable like is supported for completions via completionItem/resolve.

Anyways, I agree this is outside the scope of this PR. I'm still very much interested in improving the symbol search so it's closer to how it works in IntelliJ 😼